src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisWebhookService.java

```diff
@@ -0,0 +1,110 @@
+package de.tum.in.www1.artemis.service.connectors.pyris;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.enumeration.AttachmentType;
+import de.tum.in.www1.artemis.domain.lecture.AttachmentUnit;
+import de.tum.in.www1.artemis.service.FilePathService;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.PyrisPipelineExecutionSettingsDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.lectureIngestionWebhook.PyrisLectureUnitWebhookDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.lectureIngestionWebhook.PyrisWebhookLectureIngestionExecutionDTO;
+import de.tum.in.www1.artemis.service.iris.settings.IrisSettingsService;
+
+@Service
+@Profile("iris")
+public class PyrisWebhookService {
+
+    private static final Logger log = LoggerFactory.getLogger(PyrisWebhookService.class);
+
+    private final PyrisConnectorService pyrisConnectorService;
+
+    private final PyrisJobService pyrisJobService;
+
+    private final IrisSettingsService irisSettingsService;
+
+    @Value("${server.url}")
+    private String artemisBaseUrl;
+
+    public PyrisWebhookService(PyrisConnectorService pyrisConnectorService, PyrisJobService pyrisJobService, IrisSettingsService irisSettingsService) {
+        this.pyrisConnectorService = pyrisConnectorService;
+        this.pyrisJobService = pyrisJobService;
+        this.irisSettingsService = irisSettingsService;
+    }
+
+    private boolean lectureChatEnabled(Course course) {
+        return Boolean.TRUE.equals(irisSettingsService.getRawIrisSettingsFor(course).getIrisChatSettings().getLectureChat());
+    }
+
+    private String attachmentToBase64(AttachmentUnit attachmentUnit) {
+        Path path = FilePathService.actualPathForPublicPathOrThrow(URI.create(attachmentUnit.getAttachment().getLink()));
+        try {
+            byte[] fileBytes = Files.readAllBytes(path);
+
+            return Base64.getEncoder().encodeToString(fileBytes);
+        }
+        catch (IOException e) {
+            return e.getMessage();
+        }
+    }
+
+    private PyrisLectureUnitWebhookDTO processAttachments(Boolean shouldUpdate, AttachmentUnit attachmentUnit) {
```

Comment:
Instead of using a `shouldUpdate` param you should rather have a separate method for handling the deletion.

Reply:
I am conbining the method on the ingestion part on pyris as well, would ou rather add another endpoint, one for uploading the lectures and one for erasing them ?